internal/rpc/rpcserver/server.go

```diff
@@ -1655,6 +1655,24 @@ func (s *walletServer) PurchaseTickets(ctx context.Context,
 				"CSPP Server set, but error on changeAccount: %v", err)
 		}
 	}
+	// new vspd request
+	var vspHost string
+	var vspPubKey string
+	if req.VspHost != "" || req.VspPubkey != "" {
```

Comment:
You can use the early return to avoid these nested conditions and use `req.VspHost` and `req.VspPubKey` directly in `vsp.New()`. I found the same code in the line 2462.

Reply:
I need to check if either of the vsp params are set, that's why I need this nested if, but thanks for this comment, because it made me realize the vsp.New() was outside it, when it needed to be inside. Thanks.